src/query/test/pseudo/path.ts

```diff
@@ -0,0 +1,186 @@
+import t from 'tap'
+import { asPostcssNodeWithChildren, parse } from '@vltpkg/dss-parser'
+import { path } from '../../src/pseudo/path.ts'
+import { getPathBasedGraph } from '../fixtures/graph.ts'
+import type { ParserState } from '../../src/types.ts'
+import type { SpecOptions } from '@vltpkg/spec/browser'
+
+const specOptions = {
+  registry: 'https://registry.npmjs.org',
+  registries: {
+    custom: 'http://example.com',
+  },
+} as SpecOptions
+
+const getState = (query: string, graph = getPathBasedGraph()) => {
+  const ast = parse(query)
+  const current = ast.first.first
+  const state: ParserState = {
+    comment: '',
+    current,
+    initial: {
+      edges: new Set(graph.edges.values()),
+      nodes: new Set(graph.nodes.values()),
+    },
+    partial: {
+      edges: new Set(graph.edges.values()),
+      nodes: new Set(graph.nodes.values()),
+    },
+    collect: {
+      edges: new Set(),
+      nodes: new Set(),
+    },
+    cancellable: async () => {},
+    walk: async i => i,
+    retries: 0,
+    securityArchive: undefined,
+    specOptions,
+    signal: new AbortController().signal,
+    specificity: { idCounter: 0, commonCounter: 0 },
+    loose: false,
+  }
+  return state
+}
+
+t.test(':path selector', async t => {
```

Comment:
I think we can add a few more interesting scenarios to tests here:
- use `:path("packages/**")` should match `['a', 'b', 'y']`
- use `:path("**/a/*")` should match `['y']`